enricher/fabric8/src/main/java/io/fabric8/maven/enricher/fabric8/CamelRestDSLEnricher.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2016 Red Hat, Inc.
+ *
+ * Red Hat licenses this file to you under the Apache License, version
+ * 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package io.fabric8.maven.enricher.fabric8;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+
+import io.fabric8.maven.core.util.Configs;
+import io.fabric8.maven.core.util.MapUtil;
+import io.fabric8.maven.enricher.api.BaseEnricher;
+import io.fabric8.maven.enricher.api.EnricherContext;
+import io.fabric8.maven.enricher.api.Kind;
+
+public class CamelRestDSLEnricher extends BaseEnricher {
+    static final String ENRICHER_NAME        = "f8-camel-dsl";
+
+    //Default Domain for Service Annotations
+    static final String DEFAULT_DOMAIN       = "api.service.kubernetes.io";
```

Comment:
Aren't these the official KEPs?

https://github.com/kubernetes/community/blob/master/contributors/design-proposals/network/service-discovery.md

Reply:
@ro14nd I think it is official, look here: https://github.com/kubernetes/community/blob/master/contributors/design-proposals/network/service-discovery.md

Reply:
@KurtStam @zregvart it's not and I explained that in my comments in the google docs specs (@KurtStam where you able to read them ? Asking because I got not a single feedback yet)

In summary:


> The latest link is https://github.com/kubernetes/community/blob/master/contributors/design-proposals/network/service-discovery.md
> 
> But please note, that this is a community proposal which, according to the README might be considered to be historical "Note that a number of these documents are historical and may be out of date or unimplemented."
> 
> Nowadays, proposals are using the Kubernetes Enhancement Process (KEP) which are handled as in https://kubernetes.io/docs/community/keps/ with a defined status. The service discovery proposal didn't made it to a KEP, so if we insist of having an "official" kubernetes annotation one might consider to resubmit according to the KEP process in the appropriate SIG (netorking, I guess)
> 
> The only lists of "official" Kubernetes annotations from the kubernetes.io namespace that I found are https://kubernetes.io/docs/reference/kubernetes-api/labels-annotations-taints/ and the one mentioned within https://kubernetes.io/docs/concepts/services-networking/service/. Neither of these documents mentions the "api.service.kubernetes.io" annotations (also note that the scheme is different nowadays, with having beta.kubernetes.io style annotations for new annotations)

Also probably important:

> According to https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/ and from my experiences, too:
> 
> "You can use either labels or annotations to attach metadata to Kubernetes objects. Labels can be used to select objects and to find collections of objects that satisfy certain conditions. In contrast, annotations are not used to identify and select objects. The metadata in an annotation can be small or large, structured or unstructured, and can include characters not permitted by labels."
> 
> This means, that only labels get indexed in the K8s etcd and can be easily and effectively searched for. To find objects by annotations via the K8s API you have to iterate over all objects and check yourself. For labels afaik there is direct support for a label based query to find all services with a certain label value.
> 
> Hence I recommend using labels instead of annotations for service discovery as you not only enrich the metadata of a service but also want to search for object carrying this label.
> 
> However, the character set allowed in labels is limited, so might interfere with the possible values of the service annotations. In that case annotations would make sense again (despite the performance penalty to filter out all service by iterating over it)

Reply:
@zregvart @KurtStam here's the readme for this old community proposal repo which clarifies the status of the proposals stored in this repo.

Remember the proposal is already over 2 years old and none of these annotations appears anywhere else except in this proposal (no doc, no software). At least I pushed Google search hard but could only find this document as a reference.

Reply:
One also might say its still a 'proposal' but not a 'standard' (and won't get there except when you follow the KEP process which adds also a status to these proposals). 

I would suggest trying to submit the proposal again with the regular KEP process and in the meantime using a custom prefix.

Reply:
These are the states of a KEP --> https://kubernetes.io/docs/community/keps/#kep-workflow so we should strive to get the proposal to "implemented" state.

Keps are kept (ha, pun intended ;-) in https://github.com/kubernetes/community/tree/master/keps , the older (kind of legacy proposals) are kept in https://github.com/kubernetes/community/tree/master/contributors/design-proposals

Reply:
My interpretation is from the [readme](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/README.md) that it's an _accepted design proposal_. While we should engage in a KEP and make it to _implemented_ state I don't see what's stopping us from using these. If it's accepted design then this might just be the first implementation of it.

(2c)

Reply:
This comment seems relevant: https://github.com/kubernetes/community/issues/914#issuecomment-358995254

Reply:
@zregvart good point, I just asked directly in https://github.com/kubernetes/community/issues/914 whether these annotations can be used and whether there is kind of a central registry for official and reserved annotations.

Reply:
Nevertheless I would ommit "api.service.kubernetes.io/description-language" and "api.service.kubernetes.io/protocol" as it leads to ambiguities as the type can always infered from the mimetype of the returned descriptor, and would add something like "api.service.kubernetes.io/port" to depict a port if multiple are available.

If insisting on the description type, I would switch to a "api.service.kubernetes.io/type" reflecting a mimeType, or so.